Yet another PCES copy bug

[Full logs](http://35.247.76.217:8095/swirlds-automation/develop/4N/FCM/20231122-053919-GCP-Daily-FCM-4N/FCM-Query-2k-110m/node0001/swirlds.log)
[Test data](http://35.247.76.217:8095/swirlds-automation/develop/4N/FCM/20231122-053919-GCP-Daily-FCM-4N/FCM-Query-2k-110m/)

```
2023-11-22 06:00:16.566 2173     INFO  STATE_TO_DISK    <<scheduler signed_state_file_manager>> FileUtils: deleting temporary file due to exception
2023-11-22 06:00:16.684 2174     ERROR EXCEPTION        <<scheduler signed_state_file_manager>> SignedStateFileWriter: Exception when writing the signed state for round 1033 to disk:
java.io.UncheckedIOException: java.nio.file.NoSuchFileException: data/saved/preconsensus-events/1/2023/11/22/2023-11-22T06+00+11.041927767Z_seq185_ming8493_maxg8833_orgn0.pces
	at java.base/java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:87) ~[?:?]
	at java.base/java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:103) ~[?:?]
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132) ~[?:?]
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at com.swirlds.platform.event.preconsensus.BestEffortPreconsensusEventFileCopy.gatherPreconsensusFilesOnDisk(BestEffortPreconsensusEventFileCopy.java:223) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.BestEffortPreconsensusEventFileCopy.copyPreconsensusEventStreamFiles(BestEffortPreconsensusEventFileCopy.java:135) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.BestEffortPreconsensusEventFileCopy.lambda$copyPreconsensusEventStreamFilesRetryOnFailure$0(BestEffortPreconsensusEventFileCopy.java:91) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.io.utility.FileUtils.executeAndRename(FileUtils.java:244) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.io.utility.FileUtils.executeAndRename(FileUtils.java:220) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.event.preconsensus.BestEffortPreconsensusEventFileCopy.copyPreconsensusEventStreamFilesRetryOnFailure(BestEffortPreconsensusEventFileCopy.java:89) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.state.signed.SignedStateFileWriter.writeSignedStateFilesToDirectory(SignedStateFileWriter.java:162) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.state.signed.SignedStateFileWriter.lambda$writeSignedStateToDisk$1(SignedStateFileWriter.java:218) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.io.utility.FileUtils.executeAndRename(FileUtils.java:244) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.io.utility.FileUtils.executeAndRename(FileUtils.java:220) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.state.signed.SignedStateFileWriter.writeSignedStateToDisk(SignedStateFileWriter.java:216) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.state.signed.SignedStateFileManager.saveStateTask(SignedStateFileManager.java:177) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.platform.state.signed.SignedStateFileManager.saveStateTask(SignedStateFileManager.java:135) ~[swirlds-platform-core-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.wiring.wires.input.BindableInputWire.lambda$bind$0(BindableInputWire.java:74) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.wiring.schedulers.internal.SequentialThreadTask.handle(SequentialThreadTask.java:34) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at com.swirlds.common.wiring.schedulers.internal.SequentialThreadTaskScheduler.run(SequentialThreadTaskScheduler.java:193) ~[swirlds-common-0.45.0-SNAPSHOT.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.nio.file.NoSuchFileException: data/saved/preconsensus-events/1/2023/11/22/2023-11-22T06+00+11.041927767Z_seq185_ming8493_maxg8833_orgn0.pces
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:?]
	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148) ~[?:?]
	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) ~[?:?]
	at java.base/java.nio.file.Files.readAttributes(Files.java:1851) ~[?:?]
	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220) ~[?:?]
	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277) ~[?:?]
	at java.base/java.nio.file.FileTreeWalker.next(FileTreeWalker.java:374) ~[?:?]
	at java.base/java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:83) ~[?:?]
	... 26 more
```

--- a/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/BestEffortPreconsensusEventFileCopy.java
+++ b/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/BestEffortPreconsensusEventFileCopy.java
@@ -24,6 +24,7 @@ import com.swirlds.common.context.PlatformContext;
 import com.swirlds.common.system.NodeId;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
@@ -92,7 +93,9 @@ public final class BestEffortPreconsensusEventFileCopy {
                                 platformContext, selfId, temporaryDirectory, minimumGenerationNonAncient));
 
                 return;
-            } catch (final IOException e) {
+            } catch (final IOException | UncheckedIOException e) {
+                // Note: Files.walk() sometimes throws an UncheckedIOException (?!!), so we have to catch both.
+
                 if (triesRemaining > 0) {
                     logger.warn(STATE_TO_DISK.getMarker(), "Unable to copy PCES files. Retrying.");
                 } else {

